src/lua/api-gateway/redis/redisHealthCheck.lua

```diff
@@ -81,6 +81,12 @@ end
 -- instance may be down)
 local function isPeerHealthy(upstream, upstreamPassword)
 
+    local enableRedisAdvancedHealthcheck = ngx.var.enable_redis_advanced_healthcheck
+    if not enableRedisAdvancedHealthcheck or enableRedisAdvancedHealthcheck ~= "true" then
```

Comment:
i think enableRedisAdvancedHealthcheck ~= "true" also includes the case when enableRedisAdvancedHealthcheck is nil